pkg/types/property.go

```diff
@@ -21,7 +21,8 @@ func getPropertyTypes(schema *openapi3.SchemaRef) (map[string]Property, error) {
 			schema: v.Value.Properties,
 		}
 	}
-	if len(properties) == 0 {
+
+	if len(properties) == 0 && !(*schema.Value.AdditionalPropertiesAllowed) {
```

Comment:
since schema is being dereferenced here - please return error at the beginning of this function if schema is nil